app/src/androidTest/java/com/sdp13epfl2021/projmag/UserTypeChoiceTest.kt

```diff
@@ -0,0 +1,97 @@
+package com.sdp13epfl2021.projmag
+
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.LargeTest
+import androidx.test.rule.ActivityTestRule
+import androidx.test.runner.AndroidJUnit4
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.TypeSafeMatcher
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class UserTypeChoiceTest {
+
+    @Rule
+    @JvmField
+    var mActivityTestRule = ActivityTestRule(UserTypeChoice::class.java)
+
+    @Test
+    fun userTypeChoiceTest() {
+        val materialRadioButton = onView(
+            allOf(
+                withId(R.id.radioProfessorType), withText("Professor"),
+                childAtPosition(
+                    childAtPosition(
+                        withClassName(`is`("android.widget.RelativeLayout")),
+                        0
+                    ),
+                    0
+                ),
+                isDisplayed()
+            )
+        )
+        materialRadioButton.perform(click())
+
```

Comment:
This only tests that the button can be clicked.
It would be better to also check that it does modify the isProfessor value.
```
assertTrue(UserTypeChoice.isProfessor)
```